text/0000-inline-buildpack.md

```diff
@@ -0,0 +1,117 @@
+# Meta
+[meta]: #meta
+- Name: Inline Buildpacks
+- Start Date: 2020-06-11
+- Author(s): [Joe Kutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a type of buildpack that can be defined in the same repo as the app it is used with.
+
+# Motivation
+[motivation]: #motivation
+
+One characteristic of `Dockerfile` that Buildpacks do not replace well is the ability to create a single, one-off file to quickly define a custom build process for an app or project. A `Dockerfile` may contain just a few lines of code, and a developer can generate a complete image. The only quick way to create a custom build for an app using a buildpack is by creating no fewer than three files: `buildpack.toml`, `bin/build`, and `bin/detect` (and likely storing those in a new repo).
+
+We seek to reduce the barrier to entry for buildpacks when a developer needs to quickly create a custom, one-off build step by introducing a new mechanism that will allow a buildpack to be defined inline with an app (and with fewer files).
+
+# What it is
+[what-it-is]: #what-it-is
+
+- **inline** - existing in the same repository
+
+The target personas for this RFC are the buildpack author and buildpack user who need to create a custom, one-off build or are just getting started with buildpacks and do not want the overhead of a new repo.
+
+We propose two new keys in the `[[build.buildpacks]]` table in `project.toml`:
+
+- `inline` - a script that will be run as the `bin/build` of the inline buildpack
+- `shell` - defines the shebang line of the script in `inline`
```

Comment:
Why not let the shebang live inside the contents of `inline`?

Reply:
because @sclevine

Reply:
We talked about this in the working group, and Stephen reminded us that without the `shell`, it becomes very difficult to know how to run the script. We could require a shebang, but then we may have to parse it. We can give `shell` a default, like `bash`, and then most people won't need to use it at all.

Reply:
I'm sorry but I still don't understand. Don't we always execute `bin/build` and `bin/detect` with `bash` (per the spec). If there was a shebang in the script, wouldn't `bash` do the right thing for us?

Reply:
Previous art for `shell`: [Github Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell)